### PR TITLE
Refactor daily email logic

### PR DESCRIPTION
**Pull request by LainMcGrathTuring**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- Refactored get_article_to_send method to only make one database query and solve for when user is not following any tags
 - Added more article information to daily email 

## Related Tickets & Documents
#24, #26, #31 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
alt_text

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
- no 

## [optional] What gif best describes this PR or how it makes you feel?
alt_text

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,3 +159,40 @@ doc or email yo@dev.to.
...
+ - Added functionality to allow users to subscribe to daily DEV emails. This is found in the users notification settings. The email consists of a hot article relating to the users followed tags. If the user has no followed tags, the user will receive that days hottest article. 
+ 
+### Sample Email
+<img width="481" alt="Screen Shot 2020-03-28 at 1 04 42 PM" src="https://user-images.githubusercontent.com/19761687/77831429-c359e100-70f4-11ea-8480-f4cac7f3dda8.png">
+ 
+## Database Changes
+ - Added email_daily_dev boolean column to the Users table. Default value of this column is false. If email_daily_dev is true the user has elected into daily emails.
+ 
+ ## Wins
+ - Overflowing team positivity/moral.
```
Comment (CHaiz15) [inline README.md:181]: Ain't that the truth

File: app/labor/daily_email_logic.rb [hunk 2]
```diff
@@ -4,18 +4,21 @@ def initialize(user)
   end
 
   def get_article_to_send
-    articles_with_user_tags.
-      where.not(user: @user).
-      order(hotness_score: :desc).
-      first
+    if get_user_tag_names != []
```
Comment (CHaiz15) [inline app/labor/daily_email_logic.rb:7]: I like the way this handles the possibility that a user may not have any tags followed yet and still sends a relevant email.

File: app/labor/daily_dev.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class DailyDev
   def self.send_daily_email
     get_users.each do |user|
       article = DailyEmailLogic.new(user).get_article_to_send
-      DailyDevMailer.daily_email(user, article).deliver
+      DailyDevMailer.daily_email(user, article).deliver_now
```
Comment (CHaiz15) [inline app/labor/daily_dev.rb:5]: Nice change. I believe .deliver has been deprecated.

File: app/labor/daily_email_logic.rb (hunk 2 above)
Comment (jordanholtkamp) [reply app/labor/daily_email_logic.rb:7]: Thanks for adding this in.

**Review by jordanholtkamp — APPROVED** [review]

**Comment by jordanholtkamp** [comment]
Great work on finishing up that extra functionality Lain.

**Pull request merged**